runtime/vm/KeyHashTable.c

```diff
@@ -439,6 +441,28 @@ hashClassTableDelete(J9ClassLoader *classLoader, U_8 *className, UDATA className
 	return hashTableRemove(table, &key);
 }
 
+UDATA
+hashClassTablePackageDelete(J9VMThread *vmThread, J9ClassLoader* classLoader, J9ROMClass* romClass)
+{
+	if (isMHProxyPackage(romClass)) {
+		/* This generated package only exists for one hidden class and should be
+		* removed when its rom class is unloaded. hashClassTable package id's rely
+		* on there being a valid rom class.
+		*/
+		UDATA result = 0;
+		KeyHashTableClassEntry key;
+		key.tag = (UDATA)romClass | TAG_ROM_CLASS;
+		omrthread_monitor_enter(vmThread->javaVM->classTableMutex);
+		result = hashTableRemove(classLoader->classHashTable, &key);
+		J9UTF8* className = J9ROMCLASS_CLASSNAME(romClass);
+		Trc_VM_hashClassTablePackageDelete(vmThread, romClass, J9UTF8_LENGTH(className), J9UTF8_DATA(className));
+		omrthread_monitor_exit(vmThread->javaVM->classTableMutex);
+		return result;
+	} else {
+		return 0; // success
```

Comment:
The old C compilers dont like this style of comments, https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-comments